plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/preferences/SSLCertificatesPreference.java

```diff
@@ -0,0 +1,210 @@
+		return OpenShiftUIActivator.getDefault().getCorePreferenceStore();
+	}
+
+	/**
+	 * Called by SSLCertificateCallback to retrieve the choice made in the dialog.
+	 * 
+	 * @param certificate
+	 * @return
+	 */
+	public Boolean getAllowedByCertificate(X509Certificate certificate) {
+		X509CertificateParser p = new X509CertificateParser(certificate);
+		Item i = new Item(false, p.getIssuer(), p.getValidity(), p.getFingerprint());
+		Item item = findItem(getSavedItems(), i);
+		return item != null ? item.checked : null;
+	}
+
+	/**
+	 * Called by SSLCertificateCallback to remember the choice made in the dialog.
+	 *  
+	 * @param certificate
+	 * @return
+	 */
+	public void setAllowedByCertificate(X509Certificate certificate, boolean result) {
+		X509CertificateParser p = new X509CertificateParser(certificate);
+		Item i = new Item(result, p.getIssuer(), p.getValidity(), p.getFingerprint());
+		List<Item> savedItems = getSavedItems();
+		Item item = findItem(savedItems, i);
+		if(item == null) {
+			savedItems.add(i);
+		} else {
+			item.checked = result;
+		}
+		saveItemsToPreference(savedItems);
+	}
+
+	/**
+	 * Called by preference page.
+	 * Each item is stored as  
+	 * 	checked SEPARATOR issuedBy SEPARATOR validity SEPARATOR fingerprint SEPARATOR
+	 * 
+	 * @return the single list instance, which is never replaced, only its content is changed
+	 */
+	List<Item> getSavedItems() {
+		if(savedItems == null) {
+			List<Item> savedItems = new ArrayList<>();
+			String value = getPreferenceStore().getString(ALLOWED_CERTIFICATES);
+			if(value != null && value.length() > 0) {
+				String[] tokens = value.split(SEPARATOR);
+				for (int i = 0; i + 4 <= tokens.length;) {
+					boolean checked = "true".equals(tokens[i++]);
+					String issuedBy = tokens[i++];
+					String validity = tokens[i++];
+					String fingerprint = tokens[i++];
+					if(isValid(validity)) {
+						savedItems.add(new Item(checked, issuedBy, validity, fingerprint));
+					}
+				}
+			}
+			this.savedItems = savedItems;
+		}
+		return savedItems;
+	}
+
+	private boolean isValid(String validity) {
+		int expiresOnIndex = validity.indexOf(X509CertificateParser.EXPIRES_ON_PREFIX);
+		if(expiresOnIndex >= 0) {
+			SimpleDateFormat dateFormat = new SimpleDateFormat(X509CertificateParser.DATE_FORMAT);
+			String expiresOn = validity.substring(expiresOnIndex + X509CertificateParser.EXPIRES_ON_PREFIX.length()).trim();
+			try {
+				Date date = dateFormat.parse(expiresOn);
+				return date.getTime() > System.currentTimeMillis();
+			} catch (ParseException e) {
+				OpenShiftUIActivator.getDefault().getLogger().logError("SSLCertificatesPreference.isValid(String): Could not parse '" + expiresOn + "' in format " + X509CertificateParser.DATE_FORMAT);
+			}
+		}
+		//In case of any failure, just assume that certificate is valid.
+		return true;
+	}
+
+	private Item findItem(List<Item> items, Item item) {
+		return items.stream().filter(i -> i.equals(item)).findFirst().orElse(null);
```

Comment:
No, by advice of Fred I call this method with list returned by getSavedItems() which always returns non-null object. It is done consistently, and since findItem() is private no one else can call it with null.